src/Abstractions/TaskFailureDetails.cs

```diff
@@ -35,12 +36,37 @@ public override string ToString()
     /// </remarks>
     /// <typeparam name="T">The type of exception to test against.</typeparam>
     /// <returns>
-    /// Returns <c>true</c> if the <see cref="ErrorType"/> value matches <typeparamref name="T"/>; <c>false</c> otherwise.
+    /// <c>true</c> if the <see cref="ErrorType"/> value matches <typeparamref name="T"/>; <c>false</c> otherwise.
     /// </returns>
     public bool IsCausedBy<T>() where T : Exception
     {
+        return this.IsCausedBy(typeof(T));
+    }
+
+    /// <returns>
+    /// <c>true</c> if the <see cref="ErrorType"/> value matches <paramref name="type"/>; <c>false</c> otherwise.
+    /// </returns>
+    /// <exception cref="ArgumentException">If <paramref name="type"/> is not an exception type.</exception>
+    /// <inheritdoc cref="IsCausedBy{T}"/>
+    public bool IsCausedBy(Type type)
+    {
+        if (!typeof(Exception).IsAssignableFrom(type))
+        {
+            throw new ArgumentException($"The type {type} is not an exception type.", nameof(type));
+        }
+
+        // This check works for core .NET exception types
         this.exceptionType ??= Type.GetType(this.ErrorType, throwOnError: false);
-        return this.exceptionType != null && typeof(T).IsAssignableFrom(this.exceptionType);
+
+        // This check works for custom exception types defined in the app's assembly
+        this.exceptionType ??= Assembly.GetCallingAssembly().GetType(this.ErrorType);
+
+        // This last check works for exception types defined in any loaded assembly (e.g. NuGet packages, etc.)
+        this.exceptionType ??= AppDomain.CurrentDomain.GetAssemblies()
+            .Select(a => a.GetType(this.ErrorType, throwOnError: false))
+            .FirstOrDefault(t => t is not null);
+
+        return this.exceptionType is not null && type.IsAssignableFrom(this.exceptionType);
```

Comment:
does `this.exceptionType` being not `null` not guarantee, at this point, that it's also assignable?

Reply:
The double-negatives question might be throwing me off, but I think the answer to your question is that `this.exceptionType` not being null simply means that we successfully loaded the type definition. Once loaded, we still need to check to see if what we loaded is compatible with what the user is trying to check for.

Perhaps renaming these variables would make their purposes more clear?

Reply:
Makes sense - my misunderstanding was in that I did not fully understand the usage of `IsCausedBy`. I re-read the linked issue and this checks out.